pasqal_cloud/client.py

```diff
@@ -108,17 +109,28 @@ def _request(
         payload: Optional[Union[Mapping, Sequence[Mapping]]] = None,
         params: Optional[Mapping[str, Any]] = None,
     ) -> JSendPayload:
-        rsp = requests.request(
-            method,
-            url,
-            json=payload,
-            timeout=TIMEOUT,
-            headers={"content-type": "application/json"},
-            auth=self.authenticator,
-            params=params,
-        )
-        rsp.raise_for_status()
-        data: JSendPayload = rsp.json()
+        successful_request: bool = False
+        iteration: int = 0
+        while not successful_request and iteration <= HTTP_RETRIES:
+            # time = (interval seconds * exponent rule) ^ retries
+            delay = (1 * 2) ** iteration
+            try:
+                rsp = requests.request(
+                    method,
+                    url,
+                    json=payload,
+                    timeout=TIMEOUT,
+                    headers={"content-type": "application/json"},
+                    auth=self.authenticator,
+                    params=params,
+                )
+                data: JSendPayload = rsp.json()
+                successful_request = True
+            except Exception as e:
+                if iteration == HTTP_RETRIES:
+                    raise e
+                time.sleep(delay)
```

Comment:
Do we want to differentiate between retryable vs non-retryable error codes ? https://stackoverflow.com/questions/47680711

For example, re-sending a "bad request" is unlikely to make it work, more likely something needs to be fixed on the client side.

Reply:
Yeah I agree with this, I suppose it's a rather small pool of status codes that we want to retry for. It's mostly going to be 502s of 408s I imagine

Reply:
500s as well given the status code currently returned by our APIs